cachi2/core/package_managers/rpm/redhat.py

```diff
@@ -0,0 +1,168 @@
+import asyncio
+import hashlib
+import logging
+import uuid
+from collections import UserDict
+from os import PathLike
+from pathlib import Path
+from typing import Any, Optional, Union
+
+from pydantic import BaseModel, PositiveInt, ValidationError
+
+from cachi2.core.config import get_config
+from cachi2.core.package_managers.general import async_download_files
+
+log = logging.getLogger(__name__)
+
+
+class Package(BaseModel):
+    """Package item; represents RPM or SRPM file. Used for parsing the lockfile."""
+
+    repoid: Optional[str] = None
+    url: str
+    checksum: Optional[str] = None
+    size: Optional[int] = None
+
+
+class Arch(BaseModel):
+    """Architecture structure. Used for parsing the lockfile."""
+
+    arch: str
+    packages: list[Package] = []
+    source: list[Package] = []
+
+
+class Root(BaseModel):
+    """Top of the structure of the lockfile. Used for parsing the lockfile."""
+
+    lockfileVersion: PositiveInt
+    lockfileVendor: str
+    arches: list[Arch]
+
+
+class Header(BaseModel):
```

Comment:
Originally the `Header` was part of the format detection mechanism (code was prepared for multiple formats).
Since then the mechanism has been removed, but I kept `Header` because I was able to load only two (sufficient) items to verify I have the right format before loading the whole lockfile.

Reply:
> Since then the mechanism has been removed, but I kept Header because I was able to load only two (sufficient) items to verify I have the right format (and thus print out relevant error message) before loading the whole lockfile.

Well, we already parsed the whole YAML, so validation done only on `Header` as you propose elsewhere in the code is kinda pointless then, you can validate the whole lockfile for structure and then validate individual fields, it's what we do in the rest of the code base.